backend/src/main/java/atlantbh/restaurants/services/BaseService.java

```diff
@@ -0,0 +1,44 @@
+package atlantbh.restaurants.services;
+
+import atlantbh.restaurants.models.BaseModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public class BaseService<M extends BaseModel, R extends JpaRepository<M,Long>> {
+
+    @Autowired
+    protected R repository;
+
+    public Optional<M> getById(Long id) {
```

Comment:
This should be called findById..

Also create method:
```
public M get(pass id) {
   M model = repository.findById(id);
  if (model != null) { 
      return model;
  }
  throw new ServiceException("message");
}
```

So when you need to be sure that model for provided ID is in DB, then call get(), because it will throw service exception, and that exception will be thrown on controller. Call findById when otherwise.

Reply:
Fixed in c2ed9b4060983e1312e68ee9cdd1de2be1986207